### PR TITLE
Feature association

### PR DESCRIPTION
**Pull request by tufoinnkuo10**
In this milestone, the following was done;
- New classes for classroom, rentals, and books were created
- their different variable instances were initialized in their constructors
- setters and getters put to use
- class rental acts as the intermediate class for the many to many relationships between person and book
- git flow followed

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    @author = author
+    @rentals = []
+  end
+
+  # method for rental
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.book = self
+  end
```
Comment (Emmyn5600) [inline book.rb]: - [x] A book has many rentals, Implementing the method for `add_rental` let us pass the following parameters ``def add_rental(person, date)``

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @tufoinnkuo10 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.
      
 ## Required Changes ♻️
 
- _Check the comment under the review_
- We are missing the following implementation, please do it.

  - ` When setting the classroom for a student it also adds it to the classrooms' students:`
  
  -  ` A class Person (A person has many rentals:)`

## Optional suggestions

- N/A.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: book.rb (hunk 1 above)
Comment (tufoinnkuo10) [reply book.rb]: Thanks for the review and reminder of what i left out. the required changes have been made

**Comment by omar-labana** [comment]
Hi @tufoinnkuo10 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**